pkg/models/manifest.go

```diff
@@ -94,63 +101,66 @@ func BootstrapManifests(fluxClient flux.Flux, clusterName string, namespace stri
 
 	return []Manifest{
 		{
-			Path:    git.GetSystemQualifiedPath(clusterName, AppCRDPath),
+			Path:    git.GetSystemQualifiedPath(params.ClusterName, AppCRDPath),
 			Content: manifests.AppCRD,
 		},
 		{
-			Path:    git.GetSystemQualifiedPath(clusterName, RuntimePath),
+			Path:    git.GetSystemQualifiedPath(params.ClusterName, RuntimePath),
 			Content: runtimeManifests,
 		},
 		{
-			Path:    git.GetSystemQualifiedPath(clusterName, SystemKustResourcePath),
+			Path:    git.GetSystemQualifiedPath(params.ClusterName, SystemKustResourcePath),
 			Content: systemKustResourceManifest,
 		},
 		{
-			Path:    git.GetSystemQualifiedPath(clusterName, UserKustResourcePath),
+			Path:    git.GetSystemQualifiedPath(params.ClusterName, UserKustResourcePath),
 			Content: userKustResourceManifest,
 		},
 		{
-			Path:    git.GetSystemQualifiedPath(clusterName, WegoAppPath),
+			Path:    git.GetSystemQualifiedPath(params.ClusterName, WegoAppPath),
 			Content: wegoAppManifest,
 		},
 		{
-			Path:    git.GetSystemQualifiedPath(clusterName, WegoConfigPath),
+			Path:    git.GetSystemQualifiedPath(params.ClusterName, WegoConfigPath),
 			Content: wegoConfigManifest,
 		},
 	}, nil
 }
 
-// GitopsManifests generates all yaml files that are going to be written in the config repo
-func GitopsManifests(ctx context.Context, fluxClient flux.Flux, gitProvider gitproviders.GitProvider, clusterName string, namespace string, configURL gitproviders.RepoURL) ([]Manifest, error) {
-	bootstrapManifest, err := BootstrapManifests(fluxClient, clusterName, namespace, configURL)
-	if err != nil {
-		return nil, err
-	}
+type GitopsManifestsParams struct {
+	FluxClient    flux.Flux
+	GitProvider   gitproviders.GitProvider
+	ClusterName   string
+	WegoNamespace string
+	ConfigRepo    gitproviders.RepoURL
+}
 
-	systemKustomization := CreateKustomization(clusterName, namespace, RuntimePath, SourcePath, SystemKustResourcePath, UserKustResourcePath, WegoAppPath)
+// GitopsManifests generates all yaml files that are going to be written in the config repo
+func GitopsManifests(ctx context.Context, bootstrapManifests []Manifest, params GitopsManifestsParams) ([]Manifest, error) {
```

Comment:
small suggestion. I think we actually don't need a function that returns all the manifests at this point. I'm thinking that you probably added `bootstrapManifests` field here so we don't call BootstrapManifests twice in the Install function, which is great (Should have seen that 🤔 ). If that was you intention. I would suggest to just remove `bootstrapManifests` parameter here and just return the 3 manifests generated( `Source`, `Kustomization` and `Keep`) (and change the name of the function). Then we can just append the result of this function to the previous result from calling BootstrapManifests in `Install` method.

Reply:
No worries to much about this, I have a pending PR where I'm modifying some logic in install/cmd.go. We can address this there, once I start fixing conflicts after you merge this (I know you had a lot of conflicts to fix).

Reply:
that was exactly the case, yes please. 😄